hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/catalyst/catalog/HoodieCatalogTable.scala

```diff
@@ -245,6 +245,10 @@ class HoodieCatalogTable(val spark: SparkSession, var table: CatalogTable) exten
       case (_, false) =>
         ValidationUtils.checkArgument(table.schema.nonEmpty,
           s"Missing schema for Create Table: $catalogTableName")
+        if (sqlOptions.contains("hoodie.datasource.write.keygenerator.class") &&
+        !sqlOptions.contains("hoodie.table.keygenerator.class")) {
```

Comment:
In `HoodieSparkSqlWriter#mergeParamsAndGetHoodieConfig` we can see similar logic, we can unify the code base, also we should not use hard-code option keys.